plugin/evm/warp_service.go

```diff
@@ -0,0 +1,29 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// WarpAPI introduces snowman specific functionality to the evm
+type WarpAPI struct{ vm *VM }
+
+// GetSignature returns the BLS signature associated with a messageID. In the raw request, [messageID] should be cb58 encoded
+func (api *WarpAPI) GetSignature(ctx context.Context, messageID ids.ID) (*message.SignatureResponse, error) {
+	signature, err := api.vm.backend.GetSignature(ctx, messageID)
+	if err != nil {
+		log.Debug("Unknown warp signature requested", "messageID", messageID)
+		return nil, nil
+	}
+
+	response := message.SignatureResponse{
+		Signature: signature,
+	}
+	return &response, nil
```

Comment:
Can we change the return type here so that it will be human readable?

This currently gives the JSON response:

```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "Signature": [
            65,
            87,
            66,
            71,
            153,
            164,
            180,
            109,
            4,
            214,
            33,
            162,
            250,
            6,
            9,
            215,
            58,
            247,
            80,
            1,
            123,
            239,
            57,
            185,
            243,
            98,
            51,
            76,
            236,
            227,
            127,
            177,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0
        ]
    }
}
```

To get this I modified the code here to be:

```go
// GetSignature returns the BLS signature associated with a messageID. In the raw request, [messageID] should be cb58 encoded
func (api *WarpAPI) GetSignature(ctx context.Context, messageID ids.ID) (*message.SignatureResponse, error) {
	// signature, err := api.vm.backend.GetSignature(ctx, messageID)
	// if err != nil {
	// 	log.Debug("Unknown warp signature requested", "messageID", messageID)
	// 	return nil, nil
	// }

	var signature [96]byte
	copy(signature[:], messageID[:])

	response := message.SignatureResponse{
		Signature: signature,
	}
	return &response, nil
}
```

This should return a hex format. I would recommend wrapping the bytes with `hexutil.Bytes` as it's used in some of the other eth RPC services to wrap bytes.

Reply:
Fixed. Confirmed that the response is in hex format.